ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OAlgorithm.scala

```diff
@@ -85,4 +85,20 @@ abstract class H2OAlgorithm[P <: Model.Parameters : ClassTag] extends Estimator[
   }
 
   override def copy(extra: ParamMap): this.type = defaultCopy(extra)
+
+  protected def createKey[T <: Keyed[T]](modelId: String): Key[T] = {
+    val key = Key.make[T](modelId)
+    if(H2O.containsKey(key)) {
+      var i = 0
```

Comment:
what do you think of renaming `i` to something like `suffix`?

Reply:
Yup, no problem :-)